media/windows/ECC-Ecobee/ECC Pycobee Data.py

```diff
@@ -385,54 +427,72 @@ def main():
     print(F"*** ECC Ecobee data retrieval version {eccpycobee_version}, {eccpycobee_date} ***")
     logger.info(F"*** ECC Ecobee data retrieval version {eccpycobee_version}, {eccpycobee_date} ***")
 
+    logger.info(F"Log filename:                       {args.log_file_path}")
+    logger.info(F"Database filename:                  {args.database_file_path}")
+    logger.info(F"Authorization token filename:       {args.authorize_file_path}")
+    logger.info(F"Default API key filename:           {args.api_file_path}")
+    logger.info(F"Thermo revision interval filename:  {args.int_file_path}")
+
     # Attempt to open the credentials / authorization file and read contents
     try:
-        pyecobee_db = shelve.open(ECCAuthorize,  # read the stored authorization file
-                                  writeback=True,
-                                  protocol=4)  # protocol 4 is the latest pickle version, > Python 3.4
-        logger.debug(F"Shelve data structure:  Keys: {len(pyecobee_db)}")
-        logger.debug(F"...elements:  {list(pyecobee_db)}")
-        auth_list = list(pyecobee_db)
-        for pelement in auth_list:
-            logger.debug(F"...contents:  {pelement}: {pyecobee_db.get(pelement)}")
-
-        # Test to see if the read credentials are empty; if so, initialize the Ecobee service object.
-        # This would typically happen at first run when there are no stored credentials.
-        if len(pyecobee_db) == 0:
-            # app_key = "Vy72kTcU63iWryQ9YG5T4faruTFKxIwz"  # default, if not read from shelve
-            app_key = get_api()
-        else:
-            app_key = pyecobee_db['application_key']
+        with open(ECCAuthorize, "r") as read_auth:
+            json_auth_dict = json.load(read_auth)
+    # Handle [Errno 2] No such file or directory, JSON decoding error (syntax error in file)
+    except (FileNotFoundError, json.JSONDecodeError) as e:
+        logger.error(F"Missing or invalid authorization token JSON file...")
+        logger.error(F"...error:  {e}")
+        print(F"Missing or invalid authorization token JSON file...")
+        print(F"...error:  {e}")
```

Comment:
Quick question: I see in a bunch of places that you call both `logger.foo()` _and_ `printf`.

Is there a reason calling `logger.foo()` isn't sufficient?  FWIW, you can have `logger` output to multiple places (e.g., both a file and stdout), if that makes it easier.

Reply:
Yeah, my ignorance on the logger options here...
I'd originally directed most messages to just the log, but when running interactively, I wanted some key messages (mainly errors, but some informational messages like records written) displayed.  When running from the scheduler it doesn't matter as much.  I'd wondered if I could specify more than one output destination but hadn't pursued it.  Will check on how to do that for future code.